ui/app/components/app/signature-request-original/signature-request-original.component.js

```diff
@@ -0,0 +1,325 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import ethUtil from 'ethereumjs-util'
+import classnames from 'classnames'
+import { ObjectInspector } from 'react-inspector'
+
+import { ENVIRONMENT_TYPE_NOTIFICATION } from '../../../../../app/scripts/lib/enums'
+import { getEnvironmentType } from '../../../../../app/scripts/lib/util'
+import Identicon from '../../ui/identicon'
+import AccountListItem from '../../../pages/send/account-list-item/account-list-item.component'
+import { conversionUtil } from '../../../helpers/utils/conversion-util'
+import Button from '../../ui/button'
+import { DEFAULT_ROUTE } from '../../../helpers/constants/routes'
+
+export default class SignatureRequestOriginal extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
+    metricsEvent: PropTypes.func,
+  }
+
+  static propTypes = {
+    balance: PropTypes.any,
+    cancel: PropTypes.func,
+    clearConfirmTransaction: PropTypes.func,
+    conversionRate: PropTypes.any,
+    history: PropTypes.object,
+    requesterAddress: PropTypes.any,
+    selectedAccount: PropTypes.string,
+    sign: PropTypes.any,
+    txData: PropTypes.any,
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
```

Comment:
Can this be a property on the class?/Can this use the class property syntax?

Reply:
Then we can remove the empty ctor

Reply:
Sure! I hadn't realized you could rely upon `this.props` being set already in a class property 🤔 
I guess it works because `super(props)` needs to be first in the implied constructor, and that sets the props.